src/Service.Tests/Configuration/HotReload/ConfigurationHotReloadTests.cs

```diff
@@ -299,4 +299,36 @@ public async Task HotReloadConfigRuntimeGQLEnabledEndToEndTest()
         // Assert
         Assert.AreEqual(HttpStatusCode.NotFound, gQLResult.StatusCode);
     }
+
+    /// <summary>
+    /// Hot reload the configuration file by saving a new database type and connection string.
+    /// Validate that the response from the server is correct when making a new request after
+    /// the change in database type.
+    /// </summary>
+    [DataTestMethod]
+    [DataRow(DatabaseType.MySQL, TestCategory.MYSQL, DisplayName = "Hot-reload change to MySQL Datasource")]
+    [DataRow(DatabaseType.PostgreSQL, TestCategory.POSTGRESQL, DisplayName = "Hot-reload change to PostgreSQL Datasource")]
+    public async Task HotReloadConfigDatasourceDatabaseTypesEndToEndTest(DatabaseType expectedDatabaseType, string expectedTestCategory)
+    {
+        // Arrange
+        string expectedConnectionString = $"{ConfigurationTests.GetConnectionStringFromEnvironmentConfig(expectedTestCategory).Replace("\\", "\\\\")}";
+
+        GenerateConfigFile(
+            databaseType: expectedDatabaseType,
+            connectionString: expectedConnectionString);
+        System.Threading.Thread.Sleep(3000);
```

Comment:
Instead of adding a timer here, why dont we check for the existence of the new file, and wait until the new file is generated? Adding timer to the test makes it flaky.

Reply:
The file is already created beforehand and we are only updating the file.
We use a timer in order to ensure that the server has enough time to hot-reload before moving on with the rest of the testing.

Reply:
I understand why you are using timer, but what if the update doesnt happen within 3 sec? can we instead check for the condition you expect to be true after the expiry of the timer instead of using a timer?

Reply:
If I remember correctly, DAB is not able to work on any requests until hot-reload is finished. If that is the case, by having a request with await then the test should wait until the request and hot-reload are completed.
(Please correct me if I am wrong on that)

Reply:
@RubenCerna2079 , what happens if we don't add a sleep, does it fail?

Reply:
@abhishekkumams, yes, since it does not give enough time for the hot-reload to take place

Reply:
Created an event that triggers when hot reload is completed.
This will be used in the tests to make them wait until hot reload finishes, before continuing with the tests.